scripts/upgrade

```diff
@@ -101,42 +97,90 @@ then
+    ynh_systemd_action --service_name=nginx --action=reload
+    # Finalize the upgrade
+    ynh_local_curl "/index.php"
+    sleep 5
+    ynh_local_curl "/index.php"
+    upgrade_type="UPGRADE_APP"
+    ynh_permission_update --permission "main" --remove "visitors"
+  fi
+
+  if [[ $visitors_enabled -eq 0 ]]; then
+    ynh_systemd_action --service_name=nginx --action=reload
+    # Finalize the upgrade
+    ynh_local_curl "/index.php"
+    sleep 5
+    ynh_local_curl "/index.php"
+    upgrade_type="UPGRADE_APP"
+  fi
+fi
+#=================================================
+# UPGRADE TO 1.1.0 if necessary
+#=================================================
+
+if ynh_compare_current_package_version --comparison lt --version 1.1.0~ynh1
+then
+  ynh_script_progression --message="Upgrading upgrade to 1.1.0" --weight=5
+  # Download, check integrity, uncompress and patch the source from 0.9.8.src
+  ynh_setup_source --keep="config.local.php" --dest_dir="$final_path" --source_id="1.1.0"
+  # Set permissions on app files
+  ynh_system_user_create --username=$app
+  ynh_add_nginx_config
+  ynh_add_fpm_config --package="$extra_php_dependencies"
+  phpversion=$(ynh_app_setting_get --app=$app --key=phpversion)
+  ynh_script_progression --message="restore data..." --weight=10
+
+  backup_bdd=/tmp/association.sqlite
+  backup_squelettes=/tmp/squelettes
 
-  key=$(ynh_string_random --length=50)
-  cp -a $final_path/config.dist.php $final_path/config.local.php
-  ynh_replace_string --match_string="const SECRET_KEY = '3xUhIgGwuovRKOjVsVPQ5yUMfXUSIOX2GKzcebsz5OINrYC50r';" --replace_string="const SECRET_KEY = '$key';" --target_file="$final_path/config.local.php"
-  if [ "$path_url" == "/" ]; then
-  ynh_replace_string --match_string="//const WWW_URI = '/asso/';" --replace_string="const WWW_URI = '$path_url';" --target_file="$final_path/config.local.php"
-  else
-  ynh_replace_string --match_string="//const WWW_URI = '/asso/';" --replace_string="const WWW_URI = '$path_url/';" --target_file="$final_path/config.local.php"
+  if  [ -d "$backup_squelettes" ]
+  then
+    cp -ar $backup_squelettes $final_path/www/squelettes
   fi
+
+  if [ -e "$backup_bdd" ]
+  then
+    cp -a $backup_bdd $final_path/association.sqlite
+  fi
+
+  chown -R $app:$app $final_path
   chmod 755 $final_path
 
   # Create the visitors permission if needed
-   visitors_enabled=$(ynh_permission_has_user "main" "visitors" && echo 0 || echo 1)
-  if [[ $visitors_enabled -eq 0 ]]; then
+  visitors_enabled=$(ynh_permission_has_user "main" "visitors" && echo 0 || echo 1)
+  if [[ $visitors_enabled -eq 1 ]]; then
     ynh_permission_update --permission "main" --add "visitors"
+    ynh_systemd_action --service_name=nginx --action=reload
+    # Finalize the upgrade
+    ynh_local_curl "/index.php"
+    sleep 5
+    ynh_local_curl "/index.php"
+    upgrade_type="UPGRADE_APP"
+    ynh_permission_update --permission "main" --remove "visitors"
   fi
```

Comment:
Same here:

```suggestion
  # Temporarily enable visitors for curl if needed
  visitors_enabled=$(ynh_permission_has_user "main" "visitors" && echo yes || echo no)
  if [[ $visitors_enabled == "no" ]]; then
    ynh_permission_update --permission "main" --add "visitors"
  fi  
  # Finalize the upgrade
  ynh_local_curl "/index.php"
  sleep 5
  ynh_local_curl "/index.php"
  
  if [[ $visitors_enabled == "no" ]]; then
    ynh_permission_update --permission "main" --remove "visitors"
  fi
```

(I can't write the full suggestion because of deleted line, but the following block of code can be deleted too)

Reply:
Ok, I try this before

Reply:
I try install (on virtualbox) garradin 1.0.7 in private mode and upgrade ti 1.1 with these changes, and it seems ok ! thanks